src/include/cpu/rpp_cpu_common.hpp

```diff
@@ -4427,24 +4545,499 @@ inline RppStatus compute_bilinear_interpolation_3c_avx(__m256 *pSrcPixels, __m25
+            Rpp32s bufferLength = outputImgSize.width;
+            Rpp32s alignedLength = bufferLength &~ (kNumLanes-1);
+
+            __m128 pFirstValR = _mm_set1_ps(inRowPtrR[0]);
+            __m128 pFirstValG = _mm_set1_ps(inRowPtrG[0]);
+            __m128 pFirstValB = _mm_set1_ps(inRowPtrB[0]);
+            bool breakLoop = false;
+            int x = 0;
+            for (; x + kNumLanes <= alignedLength; x += kNumLanes)
+            {
+                __m128 pOutputChannel[kNumVecs * 3];
+                set_zeros(pOutputChannel, kNumVecs * 3);
+                __m128 *pOutputR = &pOutputChannel[0];
+                __m128 *pOutputG = &pOutputChannel[kNumVecs];
+                __m128 *pOutputB = &pOutputChannel[kNumVecs * 2];
+                for(int vec = 0, dstx = x; vec < kNumVecs; vec++, dstx += 4) // Number of dst pixels per iter
+                {
+                    Rpp32s coeffIdx = (dstx * kernelSize);
+
+                    // In case negative index values encountered the first value of the row is used
+                    if(index[dstx] < 0)
+                    {
+                        __m128i idx[4];
+                        idx[0] = _mm_set1_epi32(index[dstx]);
+                        idx[1] = _mm_set1_epi32(index[dstx + 1]);
+                        idx[2] = _mm_set1_epi32(index[dstx + 2]);
+                        idx[3] = _mm_set1_epi32(index[dstx + 3]);
+                        for(int k = 0; k < kernelSize; k += 4)
+                        {
+                            // Generate mask to determine the negative indices in the iteration
+                            __m128i indexMask[4];
+                            __m128i kVal = _mm_set1_epi32(k);
+                            __m128 pInputR[4], pInputG[4], pInputB[4], pCoeffs[4];
+                            int kernelAdd = (k + 4) > kernelSize ? kernelSize % 4 : 4;
+                            set_zeros(pInputR, 4);
+                            set_zeros(pInputG, 4);
+                            set_zeros(pInputB, 4);
+                            set_zeros(pCoeffs, 4);
+                            for(int l = 0; l < 4; l++)
+                            {
+                                indexMask[l] = _mm_cmplt_epi32(_mm_add_epi32(_mm_add_epi32(idx[l], kVal), xmm_pDstLocInit), xmm_px0);
+                                int srcx = index[dstx + l] + k;
+                                rpp_simd_load(rpp_load4_f32_to_f32, &inRowPtrR[srcx], &pInputR[l]);
+                                rpp_simd_load(rpp_load4_f32_to_f32, &inRowPtrG[srcx], &pInputG[l]);
+                                rpp_simd_load(rpp_load4_f32_to_f32, &inRowPtrB[srcx], &pInputB[l]);
+                                pCoeffs[l] = _mm_loadu_ps(&(coeffs[coeffIdx + ((l + k) * 4)]));        // Load coefficients
+                                pInputR[l] = _mm_blendv_ps(pInputR[l], pFirstValR, indexMask[l]);   // If negative index is present load the pixel value with first value in the row
```

Comment:
If indexMask is a SSE register use the same convention - pIndexMask. Also, if the values are 1 for negative index, we could more indicatively call it pNegativeIndexMask like in bilinear